zebra/zserv.c

```diff
@@ -182,13 +182,20 @@ static void zserv_encode_interface(struct stream *s, struct interface *ifp)
 static void zserv_encode_vrf(struct stream *s, struct zebra_vrf *zvrf)
 {
 	struct vrf_data data;
+	const char *ns_name = zvrf_ns_name(zvrf);
 
 	data.l.table_id = zvrf->table_id;
+
+	if (ns_name)
+		strncpy(data.ns_name, basename(ns_name), NS_NAMSIZ);
```

Comment:
This is slightly dangerous. `data.ns_name` is uninitialized so if the string returned by `basename(ns_name)` is longer than `NS_NAMSIZ` then the stack will be corrupted. I'd suggest you to use `strlcpy()` instead.

Reply:
You also need to update the `zclient_vrf_add()` function to parse the netns name you're encoding here.

Reply:
ok